test/integration/kernel.link.spec.ts

```diff
@@ -0,0 +1,580 @@
+							type: contract1.type,
+						},
+						to: {
+							id: contract2.id,
+							type: contract2.type,
+						},
+					},
+				},
+			);
+
+			const element = await ctx.kernel.getContractById(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				linkContract.id,
+			);
+
+			assert(element !== null);
+		});
+
+		it('should be able to create a link from a concrete type to the any type', async () => {
+			const createContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					id: uuid(),
+					type: 'card@1.0.0',
+				},
+			);
+
+			const supportThreadContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					id: uuid(),
+					type: 'card@1.0.0',
+				},
+			);
+
+			const relationship = await ctx.kernel.replaceContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `relationship-card-forwards-to-any`,
+					type: 'relationship@1.0.0',
+					name: 'forwards to',
+					data: {
+						inverseName: 'is forwarded by',
+						title: 'Card',
+						inverseTitle: 'Destination',
+						from: {
+							type: 'card@1.0.0',
+						},
+						to: {
+							type: '*',
+						},
+					},
+				},
+			);
+
+			assert(createContract !== null);
+			assert(supportThreadContract !== null);
+			assert(relationship !== null);
+
+			const linkContract = await ctx.kernel.insertContract(
+				ctx.logContext,
+				ctx.kernel.adminSession()!,
+				{
+					slug: `link-${createContract.slug}-forwards-to-${supportThreadContract.slug}`,
+					type: 'link@1.0.0',
+					version: '1.0.0',
+					name: 'forwards to',
+					data: {
+						inverseName: 'is forwarded by',
+						from: {
+							id: createContract.id,
+							type: createContract.type,
+						},
+						to: {
+							id: supportThreadContract.id,
+							type: supportThreadContract.type,
+						},
+					},
+				},
+			);
+			assert(linkContract !== null);
+		});
+	});
+});
+
+async function insertCardContract() {
+	return ctx.kernel.insertContract(ctx.logContext, ctx.kernel.adminSession()!, {
+		type: 'card@1.0.0',
+	});
+}
+
+async function upsertRelationshipCardIsAttachedTo() {
+	return ctx.kernel.replaceContract(
+		ctx.logContext,
+		ctx.kernel.adminSession()!,
+		{
+			slug: `relationship-card-is-attached-to-card`,
+			type: 'relationship@1.0.0',
+			name: 'is attached to',
+			data: {
+				inverseName: 'has attached element',
+				title: 'Attachment',
+				inverseTitle: 'Container',
+				from: {
+					type: 'card@1.0.0',
+				},
+				to: {
+					type: 'card@1.0.0',
+				},
+			},
+		},
+	);
+}
+
+async function upsertRelationshipIsLinkedTo() {
+	return ctx.kernel.replaceContract(
+		ctx.logContext,
+		ctx.kernel.adminSession()!,
+		{
+			slug: `relationship-card-is-linked-to-card`,
+			type: 'relationship@1.0.0',
+			name: 'is linked to',
+			data: {
+				inverseName: 'is linked to',
+				title: 'LeftCard',
+				inverseTitle: 'RightCard',
+				from: {
+					type: 'card@1.0.0',
```

Comment:
This also shouldn't be a versionioned slug, just a type slug?

Reply:
changed